[Workers] (WIP) Assets (#16610)
cloudflare/cloudflare-docs: src/content/docs/workers/static-assets/get-started.mdx

---
pcx_content_type: concept
title: Get Started
head: []
sidebar:
  order: 6
description: Run frontend websites — static or dynamic — directly on Cloudflare's global network.
---

import {
	Badge,
	Description,
	InlineBadge,
	Render,
	PackageManagers,
} from "~/components";

> This guide will instruct you through setting up and deploying a static site or a full-stack application without a framework on Workers. You may prefer to build your website from scratch if:

Reviewer: How can we find a way to have fewer words top of this page and get into it?

- You're interested in learning by implementing core functionalities on your own.

> - You're working on a simple project where a framework might be overkill.

Reviewer: ```suggestion
- You're working on a simple project where you might not need a framework
```

- You want to optimize for performance by minimizing external dependencies.

> - You require complete control over every aspect of the application.

Reviewer: What about — you want to build your own framework?

> Alternativley, you use can a number of popular frameworks(LINK) that come with ready-to-use components, a pre-defined structured architecture, and community support. View framework specific guides(LINK) to get started using a framework.

Reviewer: I think we _still_ want to lead with this as the recommended path, even here.

i.e. — most people just use frameworks — go over here.

Then — below we'll explain how to do this without a framework

## Deploy a static site

This guide will instruct you through setting up and deploying a static site on Workers.

### 1. Create a new Worker project using the CLI

[C3 (`create-cloudflare-cli`)](https://github.com/cloudflare/workers-sdk/tree/main/packages/create-cloudflare) is a command-line tool designed to help you set up and deploy new applications to Cloudflare. Open a terminal window and run C3 to create your Worker project:

<PackageManagers
	type="create"
	pkg="cloudflare@latest"
	args={"my-static-site --experimental"}
/>

<Render
	file="c3-post-run-steps"
	product="workers"
	params={{
		category: "hello-world",
		type: "Hello World - Assets-only",
		lang: "TypeScript",
	}}

> />

Reviewer: I don't know where this component lives – but it can be simplified a lot – it tells you step by step what to select, but these options can be provided directly to the initial C3 command

https://developers.cloudflare.com/pages/get-started/c3/#cli-arguments

Author: TIL this page exists. Going to keep as is for now. We should pull C3 out of pages + document `--experimental` options as a follow up

When building a full-stack application, you can use any [Workers bindings](/workers/runtime-apis/bindings/) to interact with resources on the Cloudflare developer platform.

### 1. Create a new Worker project

[C3 (`create-cloudflare-cli`)](https://github.com/cloudflare/workers-sdk/tree/main/packages/create-cloudflare) is a command-line tool designed to help you set up and deploy new applications to Cloudflare.

Open a terminal window and run C3 to create your Worker project:

<PackageManagers
	type="create"
	pkg="cloudflare@latest"
	args={"my-dynamic-site --experimental"}
/>

<Render
	file="c3-post-run-steps"
	product="workers"
	params={{
		category: "hello-world",
		type: "Hello World - Worker with Assets",
		lang: "TypeScript",
	}}
/>

### 2. Develop locally

After you have created your Worker, run the [`wrangler dev`](/workers/wrangler/commands/#dev) command in the project directory to start a local server for developing your Worker. This will allow you to preview your Worker locally during development.

```sh
npx wrangler dev
```

> ### 3. Deploy your Project

Reviewer 2: I think we need another step here of modifying the Worker to make the difference between just static assets and static assets + worker clearer
